kyo-data/shared/src/main/scala/kyo/Flat.scala

```diff
@@ -15,56 +9,23 @@ opaque type Flat[A] = Null
 object Flat:
     object unsafe:
         inline given bypass[A]: Flat[A] = null
-    end unsafe
-
-    inline given infer[A]: Flat[A] = FlatMacro.infer
-
-    @implicitNotFound("Detected nested Kyo computation '${A}', please call '.flatten' first.")
-    case class Weak[A](dummy: Null) extends AnyVal
-
-    trait WeakLowPriority:
-        inline given Weak[Nothing] = Weak.unsafe.bypass
-
-    object Weak extends WeakLowPriority:
-        inline given Weak[Unit] = unsafe.bypass
 
-        inline given [A](using inline ng: NotGiven[A <:< (Any < Nothing)]): Weak[A] = unsafe.bypass
+    inline given derive[A]: Flat[A] = FlatMacro.derive
 
-        object unsafe:
-            inline given bypass[A]: Weak[A] = Weak(null)
-    end Weak
 end Flat
 
 private object FlatMacro:
 
-    inline def infer[A]: Flat[A] = ${ macroImpl[A] }
+    inline def derive[A]: Flat[A] = ${ macroImpl[A] }
 
     def macroImpl[A: Type](using Quotes): Expr[Flat[A]] =
         import quotes.reflect.*
 
         val t = TypeRepr.of[A].dealias
 
-        object Kyo:
```

Comment:
This is an old mechanism that used to provide more detailed error messages when pending effects don't match. It isn't functional since the last major redesign so it's fine to remove.